test/unit/constraints/array.test.ts

```diff
@@ -0,0 +1,113 @@
+import "../metadataShim";
+import {registerJoi, WORKING_SCHEMA_KEY} from "../../../src/core";
+import * as Joi from 'joi';
+import { Max, Length, ArraySchema, Min, Unique } from "../../../src/constraints/array";
+import {Validator} from "../../../src/Validator";
+import { testConstraint } from "../testUtil";
+
+
+registerJoi(Joi);
+
+describe('Array Test', () => {
+    describe('Test Array methods ', () => {
+        it('Should annotate class', () => {
```

Comment:
⛏ Try to make the test descriptions as accurate as possible. A class decorator is different to a property decorator. Have a look at `boolean.test.ts`, it calls this test `should annotate the class property`.